project_2/include/utils.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "routing_packet.hpp"
+
+#include <iostream>
+#include <queue>
+
+class RoutingPacket;
+
+class PriorityQueueWrapper {
```

Comment:
What is the reason for introducing this wrapper?

Reply:
To prohibit any operations other than adding to the packet queue, only simulator should take packets from it